src/drivers/queue_interop/Queue.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+namespace yii\queue\queue_interop;
+
+use Interop\Amqp\AmqpContext;
+use Interop\Queue\PsrConnectionFactory;
+use Interop\Queue\PsrContext;
+use Interop\Queue\PsrProducer;
+use yii\base\NotSupportedException;
+use yii\queue\cli\Queue as CliQueue;
+
+class Queue extends CliQueue
+{
+    /**
+     * {@inheritdoc}
+     */
+    public $commandClass = Command::class;
+
+    /**
+     * @var string
+     */
+    public $factoryClass = null;
+
+    /**
+     * @var array
+     */
+    public $factoryConfig = [];
+
+    /**
+     * @var string
+     */
+    public $queueName = 'queue';
+
+    /**
+     * @var PsrContext
+     */
+    private $context;
+
+    /**
+     * @var PsrProducer
+     */
+    private $producer;
+
+    /**
+     * Listens queue and runs new jobs.
+     */
+    public function listen()
+    {
+        $consumer = $this->getContext()->createConsumer(
+            $this->getContext()->createQueue($this->queueName)
+        );
+
+        while (true) {
+            if ($message = $consumer->receive()) {
+                list($ttr, $body) = explode(';', $message->getBody(), 2);
+                if ($this->handleMessage(null, $body, $ttr, 1)) {
+                    $consumer->acknowledge($message);
+                }
+            }
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function pushMessage($message, $ttr, $delay, $priority)
+    {
+        if ($delay) {
+            throw new NotSupportedException('Delayed work is not supported in the driver.');
+        }
+
+        if ($priority !== null) {
+            throw new NotSupportedException('Job priority is not supported in the driver.');
+        }
+
+        if (null === $this->producer) {
+            $this->producer = $this->getContext()->createProducer();
+        }
+
+        $this->producer->send(
+            $this->getContext()->createQueue($this->queueName),
+            $this->getContext()->createMessage("$ttr;$message")
```

Comment:
Time to reserve. See guide about this:
https://github.com/yiisoft/yii2-queue/blob/master/docs/guide/retryable.md
https://github.com/yiisoft/yii2-queue/blob/master/docs/guide-ru/retryable.md